DocsScraper/src/DocsScraper.jl

```diff
@@ -0,0 +1,235 @@
+"""
+    get_base_url(url::AbstractString)
+
+Extracts the base url.
+
+# Arguments
+- `url`: The url string of which, the base url needs to be extracted
+"""
+function get_base_url(url::AbstractString)
+    parsed_url = URIs.URI(url)
+    base_url = string(parsed_url.scheme, "://", parsed_url.host,
+        parsed_url.port != nothing ? ":" * string(parsed_url.port) : "", parsed_url.path)
+    return base_url
+end
+
+"""
+    process_node!(node, heading_hierarchy::Dict, parsed_text::Vector)
+
+Process the node recursively. The function recursively calls itself for every HTML hierarchy, thereby going deeper to retrieve the text.
+
+Working:
+Since HTMML structure is complex, we need to figure out when do we insert the extracted text in `parsed_text` 
+ie., should we add the text of child hierarchy and then insert or should we insert now and let the child hierarchy make another insertion.  
+For this we employ multiple checks. If the current node is heading, directly insert into `parsed_text`. 
+If the current node is a code block, return the text inside code block with backticks. 
+If the node is neither heading nor code, then we'll need to go deeper in the hierarchy. 
+if the curretnt node is anything from the list ["p", "li", "dt", "dd", "pre", "b", "strong", "i", "cite", "address", "em", "td"]
+it is assumed that everything inside the tag is part of a single text block with inline code. 
+But when we go deeper and if there is a code block with size > 50 chars, then our assumption was false. 
+To correct this, we first insert the previously extracted text, next we insert the current code and additionally indicate the parent recursion iteration 
+that the current iteration has inserted the previously parsed text, so there is no need for parent iteration to insert the text block again. 
+We indicate this by a return flag `is_text_inserted`
+
+# Arguments
+- `node`: The root HTML node 
+- `args[1]`: heading_hierarchy - Dict used to store metadata
+- `args[2]`: parsed_text - Vector of Dicts to store parsed text and metadata
+- `args[3]`: child_new - Bool to specify if the current block (child) is part of previous block or not. 
+                        If it's not, then a new insertion needs to be created in parsed_text
+- `args[4]`: prev_text - String which contains previous text
+"""
+function process_node!(node::Gumbo.HTMLElement, args...)
```

Comment:
For readability purpose, it would be great to:
- move the implementation to a separate script file in `src/` folder
- break down this definition to make it easier to follow (ie, increase the level of abstraction)
   I'd suggest defining `process_node!` for all key node types in the logic (more modular approach and easier to follow), eg, `process_node!(type::CodeBlock(), node, ...)` vs `process_node!(type::TextBlock, node, ...)`
    Define the core block types as: `abstract type AbstractParsedBlock end; struct CodeBlock<:AbstractParsedBlock end;` etc.